tsz-compress/src/v2/decode.rs

```diff
@@ -0,0 +1,528 @@
+use crate::prelude::*;
+
+// Delta Decoding
+
+pub fn decode_i8(bits: &'_ BitBufferSlice) -> Result<([i8; 10], usize), CodingError> {
+    if bits.is_empty() {
+        return Err(CodingError::NotEnoughBits);
+    }
+    let mut decoded_buffer: [i8; 10] = [0; 10];
+    let mut decoded_buffer_index = 0;
+    let mut value: i8 = 0;
```

Comment:
The initial value should be known by decoding VLQs. I see that has not been emitted in the encoding so far, but it will need to be there.
```
A VLQ encoding of the full value for each column for the first row
A VLQ encoding of the delta from the first row to the second row
```

decode should not have separate delta and delta-delta decoding. you could encounter both in the same stream of integers. We need to keep track of the previous values, so returning after 10 elements doesn't make sense. We should be reading the whole column in one shot depending on the type of the column.